### PR TITLE
Adding a test on autoresorting on Looper threads

### PR DESCRIPTION
**Pull request by kneth**
Related to #2604 

@realm/java

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/SortTest.java [hunk 1]
```diff
@@ -16,13 +16,36 @@
...
+import java.util.concurrent.atomic.AtomicInteger;
+
 import io.realm.entities.AllTypes;
 import io.realm.internal.TableView;
+import io.realm.rule.RunInLooperThread;
+import io.realm.rule.RunTestInLooperThread;
+import io.realm.rule.TestRealmConfigurationFactory;
 
+@RunWith(AndroidJUnit4.class)
 public class SortTest extends AndroidTestCase {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/SortTest.java]: We no longer need to extends AndroidTestCase

File: realm/realm-library/src/androidTest/java/io/realm/SortTest.java [hunk 2]
```diff
@@ -33,44 +56,54 @@
...
+        context = InstrumentationRegistry.getInstrumentation().getContext();
+        realmConfig = configFactory.createConfiguration();
+        realm = Realm.getInstance(realmConfig);
+
+        populateRealm(realm);
     }
 
-    @Override
+    @After
     public void tearDown() throws Exception {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/SortTest.java]: Remove throws Exception

File: realm/realm-library/src/androidTest/java/io/realm/SortTest.java [hunk 3]
```diff
@@ -195,108 +228,198 @@ private void checkSortTwoFieldsStringAscendingIntDescending(RealmResults<AllType
...
         } catch (IllegalArgumentException ignored) {
         }
     }
+
+    @Test
+    @RunTestInLooperThread
+    public void resorting() throws InterruptedException {
+        final AtomicInteger changeListenerCalled = new AtomicInteger(4);
+
+
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/SortTest.java]: To much widespace

**Comment by cmelchior** [comment]
Apart from a few style nitpicks  👍

**Comment by zaki50** [comment]
:+1:

**Pull request merged**